pkg/util/errors/errors.go

```diff
@@ -4,17 +4,40 @@ import (
 	"net"
 	"net/url"
 	"os"
+	"strings"
 	"syscall"
 
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/azure"
 )
 
+// some common network errors that we check for
+var str2Syscall = map[string]syscall.Errno{
+	"connection reset by peer":  syscall.ECONNRESET,
+	"connection refused":        syscall.ECONNREFUSED,
+	"network is unreachable":    syscall.ENETUNREACH,
+	"no such file or directory": syscall.ENOENT,
+}
+
+func unwrapNetHTTPError(err error) (bool, error) {
+	if !strings.HasPrefix(err.Error(), "net/http: HTTP/1.x transport connection broken:") {
+		return false, err
+	}
+	for sysMsg, errno := range str2Syscall {
+		if strings.Contains(err.Error()[47:], sysMsg) {
```

Comment:
Seems like
```
if strings.Contains(err.Error(), sysMsg) {
```
would be enough here since we assume the string will be constantly starting the same (HasPrefix) and doesn't conflict with the strings we compare with?

Reply:
absolutely, I just wanted to save some searching time. if that is not worth it, i can remove it. it is in the error path so performance not really a priority

Reply:
that's right, clearer is better than faster here, but it's optional.

Reply:
/lgtm

Reply:
i have cleaned it up a bit, i hope it's better to read now..